brmp/backend.py

```diff
@@ -113,3 +116,22 @@ def apply_default_hmc_args(iter, warmup, num_chains):
     warmup = iter // 2 if warmup is None else warmup
     num_chains = 1 if num_chains is None else num_chains
     return iter, warmup, num_chains
+
+# Helpers for working with arrays of samples. These work with both
+# numpy and torch arrays.
+
+# (num_chains, num_samples, ...) -> (num_chains * num_samples, ...)
+def flatten(arr):
+    return arr.reshape((arr.shape[0] * arr.shape[1],) + arr.shape[2:])
```

Comment:
nit: can be simplifed to `arr.reshape((-1,) + arr.shape[2:])`.

Reply:
I thought that originally too, but in this case `arr` can have shape e.g. `(1,1,0)`, in which case reshaping to `(-1,0)` is ambiguous.

Reply:
Ahh..interesting, I didn't realize that. Thanks for explaining.